core/backend/src/test/standalone/IModelTransformer.test.ts

```diff
@@ -958,4 +958,144 @@ describe("IModelTransformer", () => {
   });
 
+  it("handles definition element scoped by non-definitional element (PR 1813)", async () => {
+    const sourceDbPath = IModelTestUtils.prepareOutputFile("IModelTransformer", "BadPredecessorsExampleSource.bim");
+    const sourceDb = SnapshotDb.createEmpty(sourceDbPath, { rootSubject: { name: "BadPredecessorExampleSource" } });
+    const requestContext = new BackendRequestContext();
+
+    // create a document partition in our iModel's root
+    const documentListModelId = DocumentListModel.insert(sourceDb, IModelDb.rootSubjectId, "DocumentList");
+
+    // add a drawing to the document partition's model
+    const drawingId = sourceDb.elements.insertElement({
+      classFullName: Drawing.classFullName,
+      model: documentListModelId,
+      code: Drawing.createCode(sourceDb, documentListModelId, "Drawing"),
+    });
+    expect(Id64.isValidId64(drawingId)).to.be.true;
+
+    // submodel our drawing with a DrawingModel
+    const model = sourceDb.models.createModel({
+      classFullName: DrawingModel.classFullName,
+      modeledElement: { id: drawingId },
+    });
+    sourceDb.models.insertModel(model);
+
+    const myCodeSpecId = sourceDb.codeSpecs.insert(CodeSpec.create(sourceDb, "MyCodeSpec", CodeScopeSpec.Type.RelatedElement));
+
+    // insert a definition element which is scoped by a non-definition element (the drawing)
+    const _physicalMaterialId = sourceDb.elements.insertElement({
+      classFullName: GenericPhysicalMaterial.classFullName,
+      model: IModel.dictionaryId,
+      code: new Code({spec: myCodeSpecId, scope: drawingId, value: "physical material"}),
+    } as DefinitionElementProps);
+
+    sourceDb.saveChanges();
+
+    const targetDbPath = IModelTestUtils.prepareOutputFile("IModelTransformer", "BadPredecessorExampleTarget.bim");
+    const targetDb = SnapshotDb.createEmpty(targetDbPath, { rootSubject: { name: sourceDb.rootSubject.name } });
+    const transformer = new IModelTransformer(sourceDb, targetDb);
+
+    await expect(transformer.processSchemas(requestContext)).to.eventually.be.fulfilled;
+    await expect(transformer.processAll()).to.eventually.be.fulfilled;
+
+    // check if target imodel has the elements that source imodel had
+    expect(targetDb.codeSpecs.hasName("MyCodeSpec")).to.be.true;
+    const drawingIdTarget = targetDb.elements.queryElementIdByCode(Drawing.createCode(targetDb, documentListModelId, "Drawing"));
+    expect(drawingIdTarget).to.not.be.undefined;
+    expect(Id64.isValidId64((drawingIdTarget as string))).to.be.true;
+    const physicalMaterialIdTarget = targetDb.elements.queryElementIdByCode(new Code({spec: myCodeSpecId, scope: drawingId, value: "physical material"}));
+    expect(physicalMaterialIdTarget).to.not.be.undefined;
+    expect(Id64.isValidId64((physicalMaterialIdTarget as string))).to.be.true;
+
+    sourceDb.close();
+    targetDb.close();
+  });
+
+  it("handle backwards element insertion on models not pre-loading scope (PR 1813)", async () => {

```

Comment:
can you change the name of this test (I know I originally named it) to `handle backwards related-instance code in model`, and maybe remove the `(PR 1813)` from each test name?